event_registration_seat_limit/models.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# © 2015 Grupo ESOC Ingeniería de Servicios, S.L.U.
+
+from openerp import api, fields, models
+from . import exceptions
+
+
+class Event(models.Model):
+    """Events enhaced with participants per registration limits."""
+    _inherit = "event.event"
+
+    registration_seats_max = fields.Integer(
+        "Maximum of participants per registration",
+        default=0,
+        help="Registrations with more than this number of participants will "
+             "be forbidden. Set 0 to ignore this setting.")
+
+    registration_seats_min = fields.Integer(
+        "Minimum of participants per registration",
+        default=1,
+        help="Registrations with less than this number of participants will "
+             "be forbidden.")
+
+    @api.multi
+    @api.constrains("registration_seats_max",
+                    "registration_seats_min",
+                    "seats_max")
+    def _check_seats_per_registration_limits(self):
+        """Ensure you are not setting an invalid maximum."""
+        for s in self:
+            if s.registration_seats_min < 1:
+                raise exceptions.NeedAtLeastOneParticipant()
+
+            if s.registration_seats_max:
+                if (s.registration_seats_max <
+                        s.registration_seats_min):
+                    raise exceptions.MaxSmallerThanMin()
+
+            if s.seats_max:
+                if s.registration_seats_max > s.seats_max:
+                    raise exceptions.MaxPerRegisterBiggerThanMaxPerEvent()
+
+            try:
+                s.registration_ids._check_seats_per_registration_limits()
+            except exceptions.SeatsPerRegistrationError as error:
+                raise exceptions.PreviousRegistrationsFail(error)
```

Comment:
Error message from `SeatsPerRegistrationError` can be confusing when raised from here. `PreviousRegistrationsFail` makes that message easier to understand.